[12.0][MIG] connector_magento

sebalix opened the pull request:
PR itself ready to be reviewed, but dependencies are not ready.

Depends on:

- <s>base_technical_user: https://github.com/OCA/server-tools/pull/1560</s>
- <s>connector_ecommerce: https://github.com/OCA/connector-ecommerce/pull/49</s>
    - <s>sale_exception: https://github.com/OCA/sale-workflow/pull/831</s>
        - <s>base_exception: https://github.com/OCA/server-tools/pull/1557</s>
    - <s>sale_automatic_workflow_payment_mode: https://github.com/OCA/sale-workflow/pull/786</s>
        - <s>sale_automatic_workflow: https://github.com/OCA/sale-workflow/pull/830</s>
        - <s>[BUG] account_payment_sale: https://github.com/OCA/bank-payment/issues/536</s>
          fixed by https://github.com/OCA/bank-payment/pull/630/files

`oca_dependencies.txt` has been updated to include all the dependencies and install the module on Travis.

<s>Documentation generation is stuck because of this bug: https://github.com/odoo/odoo/issues/28125
Don't know what to do about this, excepting disable the `api/api_models` part of the documentation (need to check if some other parts depend on it).</s>
EDIT: I disabled the documentation for class members using `fields` as a workaround.

Commit:
Rename magentoerpconnect to connector_magento

Commit:
Restructure files, split large python files

Commit:
Fix views

* Add binding on sales orders
* Ensure that we always have the Connector Manager group on the
  Connector tabs (most are done in the parent views of
  connector_ecommerce)
* Fixed inheritance of the delivery.carrier view

Commit:
Collection.work_on is now a context manager

Commit:
Add a base listener and use it in partner export

Commit:
Rewrite tests using new API and vcr.py

* Use vcr.py to record the exchanges instead of a self-made recorder
* Use new base Components test case classes
* Separate export tests in 2 phases: trigger of the export which check
  that the jobs are delayed, and test the job itself in a second test

Commit:
Update documentation

Commit:
Various fixes

Commit:
Add dependency on product_multi_category

I removed it earlier, but it is required when we import products

Commit:
Manage multi-company backend

Commit:
Fix Pylint warnings

Commit:
Remove deprecated select for index creation

Commit:
[FIX] pass http credentials to backedn component if required

Commit:
[FIX] change user on backend otherwise import sale order will be done on the wrong one

Commit:
[CHG]review product aggregation by backend to improve performance

Commit:
Apply fiscal position

Commit:
Remove related_action from import_batch

jaredkipe committed:
Allow service type product based on 'giftcard' type in Magento.

jaredkipe committed:
Add Magento Enterprise Gift Cards payment line.

Unpacks the giftcard code from the PHP serialized information about the giftcards using regex.

jaredkipe committed:
Don't add a 0 amount Giftcard line.

Since the value will be a string like '0.0000', the prior tests were insufficient to prevent always adding a line.

jaredkipe committed:
Add Rewards and StoreCredit payment methods.

(Travis Bump)

Commit:
Change return value matching with the comment

Commit:
Fix failing test

* Rename it because it was shadowing another test
* Compute in the location's context

jaredkipe committed:
Listen for new `on_picking_dropship_done` event.

lmignon committed:
Add new wizard to read data from magento
The wizard is available on the backend.

oca-transbot committed:
OCA Transbot updated translations from Transifex

hugosantosred committed:
[MIG] connector_magento to v11

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update connector_magento.pot

oca-travis committed:
[UPD] Update connector_magento.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: connector-magento-10.0/connector-magento-10.0-connector_magento
Translate-URL: https://translation.odoo-community.org/projects/connector-magento-10-0/connector-magento-10-0-connector_magento/

sebalix committed:
[MIG] connector_magento to v11

sebalix committed:
[MIG] connector_magento: Migration to 12.0

pedrobaeza commented:
This is missing all the commit history in 10.0, as you are using a different starting branch.

pedrobaeza commented:
Well, not all, but a part. Maybe it was added after the 11.0 PR. Please compare both.

sebalix commented:
@pedrobaeza indeed, I tried to retrieve all previous commits when the module was still called `magentoerpconnect` but didn't succeed (doing two git format-patch, one for `magentoerpconnect` and one for `connector_magento`). I'll try again.

pedrobaeza commented:
Well, I think it's enough with commits after renaming. I miss 2 o 3 only.

sebalix committed:
[FIX] connector_magento: disable the documentation of fields

This bug https://github.com/odoo/odoo/issues/28125 is blocking the
generation of the fields documentation in 'doc/api/api_model.rst'.

gdgellatly commented:
@sebalix I closed the sale_automatic_workflow PR after encountering serious issues in production.

sebalix commented:
@gdgellatly thank you for the feedback. If you have some traceback or info to share about these issues we are interested ;)

sebalix committed:
[FIX] connector_magento: _apply_on may be None or a list of data models

bealdav commented:
Hi @sebalix Thanks for this contribution.
I saw this connector we'll not be compatible with magento 2.

Do you know any initiative on Magento 2 connection at least on the product part ?

Thanks

sebalix commented:
Hi @bealdav, sorry for the delay, yes @StefanRijnhart has worked on it for Odoo 8 (https://github.com/OCA/connector-magento/issues/211), and we plan to migrate his work on Odoo 12 (I'll probably start to work on that the next week).

bealdav commented:
Thanks for your reply

sebalix commented:
PR is now green but we are still missing the `connector-ecommerce` in 12.0.

rvalyi commented:
Hello @sebalix https://github.com/OCA/connector-ecommerce/pull/49 has been merged yesterday. How is this Magento PR now? I'm doing tests here but not in production yet...

sebalix commented:
@rvalyi I rebased this branch to remove the pending merge on https://github.com/OCA/connector-ecommerce/pull/49, it's ready to review/merge now.

JordiBForgeFlow commented:
@sebalix hi! Have you evaluated the complexity to support Magento 2?

sebalix commented:
@JordiBForgeFlow Hi! Not on latest versions of Odoo, but we are currently migrating this connector to Magento 2.2+ for OpenERP 7.0... :) (based on the work already started by @StefanRijnhart on Odoo 8.0 here: https://github.com/OCA/connector-magento/pull/289 ). I should take time to forward port some fixes in his PR BTW.
But nothing complex, a lot of code is reusable as-is, we mainly need to split ConnectorUnit/Component for each version to subclass a common one.

JordiBForgeFlow commented:
@sebalix Have you seen this?

https://github.com/MindAndGo/connector-magento/tree/10.0-callino-rebase
https://github.com/wpichler/connector-magento/tree/10.0-callino-rebase

StefanRijnhart commented on connector_magento/tests/fixtures/cassettes/:w:
```diff
@@ -0,0 +1,163 @@
+interactions:
```
I think this file was added unintendedly.

StefanRijnhart commented:
Great work! I've got just the one comment.

StefanRijnhart commented:
@sebalix please throw your fixes in my direction. I'll be starting to port Magento2 compatibility to 12.0 based to this branch if noone else is on it already (@JordiBForgeFlow?)

StefanRijnhart committed:
!fixup Were you trying to exit Vim?

StefanRijnhart commented:
@sebalix I took the liberty of removing this ':w' file myself and pushing it to your branch because that is the only flaw I could find this PR. Hoping we can merge it soon.

StefanRijnhart commented:
Oh, I see runbot fails. Probably on these warnings:
```
2020-04-04 08:50:16,654 170 WARNING openerp_test odoo.models: The model magento.config.specializer has no _description
2020-04-04 08:50:16,660 170 WARNING openerp_test odoo.models: The model magento.binding.backend.read has no _description
2020-04-04 08:50:17,948 170 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (company, company_id) of res.partner() have the same label: Company.
2020-04-04 08:50:17,994 170 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (company_id, company) of magento.res.partner() have the same label: Company.
2020-04-04 08:50:18,054 170 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (company_id, company) of magento.address() have the same label: Company.
2020-04-04 08:50:18,148 170 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (product_variant_id, odoo_id) of magento.product.product() have the same label: Product.
2020-04-04 08:50:18,555 170 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (company, company_id) of res.users() have the same label: Company.
2020-04-04 08:50:19,167 170 WARNING openerp_test odoo.models: exception.rule.create() with unknown fields: rule_group
2020-04-04 08:50:19,173 170 WARNING openerp_test odoo.models: exception.rule.create() with unknown fields: rule_group
2020-04-04 08:50:19,957 170 WARNING openerp_test odoo.addons.base.models.ir_ui_view: The group base.group_multi_salesteams defined in view magento.storeview.form does not exist!
```

sebalix commented:
@StefanRijnhart thank you for fixing this branch! Feel free to push on my branch to get it merged, I'm quite busy at the moment so I can't work on it sadly.

StefanRijnhart commented:
Thanks, I'll have a look at the warnings then.

StefanRijnhart committed:
!fixup installation warnings

StefanRijnhart committed:
!fixup no traceback logging

StefanRijnhart commented:
Runbot green! Travis borked on failed nodejs download (and feedback missing in Github?) but previous run was green: https://travis-ci.org/github/OCA/connector-magento/builds/671728251

StefanRijnhart commented:
@sebalix ~~on your review I will squash my fixes into your migration commit~~Guessing I will be prevented from force pushing the branch so please go ahead with squashing my fixes into your commit yourself.

sebalix commented on connector_magento/tests/fixtures/cassettes/:w:
```diff
@@ -1,163 +0,0 @@
-interactions:
```
:D

sebalix reviewed (commented):
@StefanRijnhart I reviewed your commit, LG (but can't approve my own PR), feel free to squash your commits

sebalix commented:
@JordiBForgeFlow a bit late sorry, no sadly I didn't see these branches until now.

StefanRijnhart commented:
@sebalix thanks for the review. Can you squash my little commits into your migration commit?

StefanRijnhart commented on connector_magento/components/importer.py:
```diff
@@ -0,0 +1,324 @@
+        for storeview in lang_storeviews:
+            lang_record = self._get_magento_data(storeview.external_id)
+            map_record = mapper.map_record(lang_record)
+            record = list(map_record.values())
```
Don't convert this dictionary of values to `list`. You can tell from line 320 that `record` needs to stay in dictionary format so that you can fetch its `items()`

StefanRijnhart commented on connector_magento/models/stock_picking/exporter.py:
```diff
@@ -0,0 +1,99 @@
+        item_qty = {}
+        # get product and quantities to ship from the picking
+        for line in binding.move_lines:
+            sale_line = line.procurement_id.sale_line_id
```
`sale_line = line.sale_line_id`

pedrobaeza commented:
@sebalix do you accept this to be superseded by #304 ?

sebalix commented:
@StefanRijnhart @pedrobaeza sorry for the delay, yes of course, I'll close this one.

StefanRijnhart commented:
Thanks, and thank you for your work on this @sebalix!
